<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingDialog</class>
 <widget class="QDialog" name="SettingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>594</width>
    <height>508</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QTabWidget" name="tab_widget_settings">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_gui">
      <attribute name="title">
       <string>Gui</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="group_box_gui_basic">
         <property name="title">
          <string>Basic</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QCheckBox" name="check_box_exit_confirmation">
            <property name="text">
             <string>Exit confirmation</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontal_layout_gui_basic_style">
            <item>
             <widget class="QLabel" name="label_gui_style">
              <property name="text">
               <string>Style</string>
              </property>
              <property name="buddy">
               <cstring>combo_box_gui_style</cstring>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="combo_box_gui_style"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_box_gui_rom_folders">
         <property name="title">
          <string>Rom Folders</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QListView" name="list_view_rom_folders">
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_gui_rom_folders_buttons">
            <item>
             <widget class="QPushButton" name="push_button_rom_folders_add">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="push_button_rom_folders_remove">
              <property name="text">
               <string>Remove</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="push_button_rom_folders_clear">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_emu">
      <attribute name="title">
       <string>Emu</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QGridLayout" name="grid_layout_emu">
         <item row="0" column="0">
          <widget class="QLabel" name="label_emu_engine">
           <property name="text">
            <string>Engine</string>
           </property>
           <property name="buddy">
            <cstring>combo_box_emu_engine</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="combo_box_emu_engine"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_emu_render_driver">
           <property name="text">
            <string>Render driver</string>
           </property>
           <property name="buddy">
            <cstring>combo_box_emu_engine</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="combo_box_emu_render_driver"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="group_box_emu_video">
         <property name="title">
          <string>Video</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QCheckBox" name="check_box_emu_full_screen">
            <property name="text">
             <string>Full screen</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="check_box_emu_vsync">
            <property name="text">
             <string>Vertical sync</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_emu_scale">
            <property name="text">
             <string>Scale</string>
            </property>
            <property name="buddy">
             <cstring>double_spin_box_emu_scale</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="double_spin_box_emu_scale">
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>15.000000000000000</double>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="double_spin_box_emu_fps">
            <property name="maximum">
             <double>60.000000000000000</double>
            </property>
            <property name="value">
             <double>60.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_emu_fps">
            <property name="text">
             <string>FPS</string>
            </property>
            <property name="buddy">
             <cstring>double_spin_box_emu_fps</cstring>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_box_emu_audio">
         <property name="title">
          <string>Audio</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="1">
           <widget class="QLabel" name="label_emu_volume">
            <property name="text">
             <string>Volume</string>
            </property>
            <property name="buddy">
             <cstring>double_spin_box_emu_fps</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_emu_sample_rate">
            <property name="text">
             <string>Sample rate</string>
            </property>
            <property name="buddy">
             <cstring>double_spin_box_emu_scale</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QSpinBox" name="spin_box_emu_sample_rate">
            <property name="minimum">
             <number>22050</number>
            </property>
            <property name="maximum">
             <number>96000</number>
            </property>
            <property name="value">
             <number>44100</number>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QSlider" name="horizontal_slider_emu_volume">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="sliderPosition">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="group_box_emu_custom">
         <property name="title">
          <string>Custom</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_emu_tape_length">
            <property name="text">
             <string>Tape length</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spin_box_emu_tape_length">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>8192</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_emu_sprite_limit">
            <property name="text">
             <string>Sprite Limit</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="spin_box_emu_sprite_limit">
            <property name="minimum">
             <number>8</number>
            </property>
            <property name="maximum">
             <number>16</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_palette">
      <attribute name="title">
       <string>Palette</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <widget class="QGroupBox" name="group_box_palette_custom">
         <property name="title">
          <string>Custom</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QHBoxLayout" name="horizontal_layout_palette_custom">
            <item>
             <widget class="QPushButton" name="push_button_palette_save">
              <property name="text">
               <string>Save</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontal_spacer_palette_custom">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="push_button_palette_load">
              <property name="text">
               <string>Load</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="push_button_palette_path_clear">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLineEdit" name="line_edit_palette_path"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="group_box_palette_control">
         <property name="title">
          <string>Control</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_input">
      <attribute name="title">
       <string>Input</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QGroupBox" name="group_box_input_port1">
         <property name="title">
          <string>Port 1</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QPushButton" name="push_button_input_port1_configure">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="combo_box_input_port1_joypad_type"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="groupBox_input_port2">
         <property name="title">
          <string>Port 2</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QPushButton" name="push_button_input_port2_configure">
            <property name="text">
             <string>Configure</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="combo_box_input_port2_joypad_type"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_input_presets">
         <property name="title">
          <string>Presets</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QPushButton" name="push_button_input_presets_save">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="push_button_input_presets_load">
            <property name="text">
             <string>Load</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontal_layout_setting_buttons">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="push_button_setting_close">
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>push_button_setting_close</sender>
   <signal>clicked()</signal>
   <receiver>SettingDialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>push_button_palette_path_clear</sender>
   <signal>clicked()</signal>
   <receiver>line_edit_palette_path</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
